tools/cli/workflowCommands.go

```diff
@@ -297,29 +313,33 @@ func processSearchAttr(c *cli.Context) *commonpb.SearchAttributes {
 	if len(searchAttrKeys) == 0 {
 		return nil
 	}
-	searchAttrVals := sanitize(c.String(FlagSearchAttributesVal))
-	if len(searchAttrVals) == 0 {
+	rawSearchAttrVals := sanitize(c.String(FlagSearchAttributesVal))
+	if len(rawSearchAttrVals) == 0 {
 		return nil
 	}
 
-	if len(searchAttrKeys) != len(searchAttrVals) {
-		ErrorAndExit(fmt.Sprintf("Uneven number of search attributes keys (%d): %v and values(%d): %v.", len(searchAttrKeys), searchAttrKeys, len(searchAttrVals), searchAttrVals), nil)
+	if len(searchAttrKeys) != len(rawSearchAttrVals) {
+		ErrorAndExit(fmt.Sprintf("Uneven number of search attributes keys (%d): %v and values(%d): %v.", len(searchAttrKeys), searchAttrKeys, len(rawSearchAttrVals), rawSearchAttrVals), nil)
 	}
 
-	searchAttributesStr := make(map[string]string, len(searchAttrKeys))
-	for i, searchAttrVal := range searchAttrVals {
-		searchAttributesStr[searchAttrKeys[i]] = searchAttrVal
-	}
+	var searchAttrValues []interface{}
 
-	searchAttributes, err := searchattribute.Parse(searchAttributesStr, nil)
```

Comment:
So you don't want to validate search attributes on the CLI side anymore. You just unmarshal them using json (this is a contract between CLI user and CLI) and send them to server. Server will do all names and types validation and return an error if it fails. I think this is a good approach.